.circleci/config.yml

```diff
@@ -23,6 +23,20 @@ jobs:
     - image: python:3.7
     steps:
     - checkout
+    - run: &skip_unmodified
+        name: Early return when dir not modified
+        command: |
+          if [ "$CIRCLE_BRANCH" = master ]; then
+              echo "Run tests because branch is master"
+          elif git log --format=%B --no-merges -n 1 | grep -qF '[run-tests]'; then
+              echo "Run tests because [run-tests] in commit message"
+          elif ! git diff --quiet origin/master... -- .; then
+              echo "Run tests because $(git rev-parse --show-prefix) was modified since branching off master"
```

Comment:
Should we add one more check that runs all tests if `.circleci` is modified, since that could have an effect on whether tests will succeed or fail?